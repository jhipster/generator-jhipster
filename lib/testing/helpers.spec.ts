import { before, describe, expect, it } from 'esmocha';
import { defaultHelpers as helpers, runResult } from './helpers.js';

const DUMMY_NAMESPACE = 'jhipster:dummy';

describe('helpers', () => {
  describe('run defaults', () => {
    before(async () => {
      await helpers.run(helpers.createDummyGenerator(), { namespace: DUMMY_NAMESPACE });
    });
    it('should register not jhipster generators namespaces', () => {
      expect(
        Object.keys((runResult.env as any).store._meta)
          .filter(ns => ns !== DUMMY_NAMESPACE)
          .sort(),
      ).toHaveLength(0);
    });
  });
  describe('runJHipster defaults', () => {
    before(async () => {
      await helpers.runJHipster('dummy').withGenerators([[helpers.createDummyGenerator(), { namespace: DUMMY_NAMESPACE }]]);
    });
    it('should register jhipster generators namespaces', () => {
      expect(
        Object.keys((runResult.env as any).store._meta)
          .filter(ns => ns !== DUMMY_NAMESPACE)
          .sort(),
      ).toMatchSnapshot();
    });
  });
  describe('run using withJHipsterGenerators', () => {
    before(async () => {
      await helpers.run(helpers.createDummyGenerator(), { namespace: DUMMY_NAMESPACE }).withJHipsterGenerators();
    });
    it('should register jhipster generators namespaces', () => {
      expect(
        Object.keys((runResult.env as any).store._meta)
          .filter(ns => ns !== DUMMY_NAMESPACE)
          .sort(),
      ).toMatchSnapshot();
    });
  });
  describe('runJHipster with useDefaultMocks', () => {
    before(async () => {
      await helpers
        .runJHipster('dummy', { useDefaultMocks: true })
        .withGenerators([[helpers.createDummyGenerator(), { namespace: DUMMY_NAMESPACE }]]);
    });
    it('should register jhipster generators namespaces', () => {
      expect(
        Object.keys((runResult.env as any).store._meta)
          .filter(ns => ns !== DUMMY_NAMESPACE)
          .sort(),
      ).toMatchSnapshot();
    });
  });
});
