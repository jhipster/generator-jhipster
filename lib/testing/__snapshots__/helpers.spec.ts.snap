// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers run using withJHipsterGenerators should register jhipster generators namespaces 1`] = `
[
  "jhipster:angular",
  "jhipster:app",
  "jhipster:base",
  "jhipster:base-application",
  "jhipster:base-core",
  "jhipster:base-entity-changes",
  "jhipster:base-workspaces",
  "jhipster:bootstrap",
  "jhipster:bootstrap-application",
  "jhipster:bootstrap-application-base",
  "jhipster:bootstrap-application-client",
  "jhipster:bootstrap-application-server",
  "jhipster:bootstrap-workspaces",
  "jhipster:ci-cd",
  "jhipster:client",
  "jhipster:client:common",
  "jhipster:common",
  "jhipster:cucumber",
  "jhipster:cypress",
  "jhipster:docker",
  "jhipster:docker-compose",
  "jhipster:entities",
  "jhipster:entity",
  "jhipster:export-jdl",
  "jhipster:feign-client",
  "jhipster:gatling",
  "jhipster:generate-blueprint",
  "jhipster:git",
  "jhipster:gradle",
  "jhipster:gradle:code-quality",
  "jhipster:gradle:jib",
  "jhipster:gradle:node-gradle",
  "jhipster:heroku",
  "jhipster:info",
  "jhipster:init",
  "jhipster:java",
  "jhipster:java:bootstrap",
  "jhipster:java:build-tool",
  "jhipster:java:code-quality",
  "jhipster:java:domain",
  "jhipster:java:jib",
  "jhipster:java:node",
  "jhipster:java:openapi-generator",
  "jhipster:javascript:bootstrap",
  "jhipster:javascript:eslint",
  "jhipster:javascript:husky",
  "jhipster:javascript:prettier",
  "jhipster:jdl",
  "jhipster:kubernetes",
  "jhipster:kubernetes-helm",
  "jhipster:kubernetes-knative",
  "jhipster:languages",
  "jhipster:liquibase",
  "jhipster:maven",
  "jhipster:maven:code-quality",
  "jhipster:maven:frontend-plugin",
  "jhipster:maven:jib",
  "jhipster:project-name",
  "jhipster:react",
  "jhipster:server",
  "jhipster:spring-boot",
  "jhipster:spring-cache",
  "jhipster:spring-cloud-stream",
  "jhipster:spring-cloud-stream:kafka",
  "jhipster:spring-cloud-stream:pulsar",
  "jhipster:spring-cloud:gateway",
  "jhipster:spring-data-cassandra",
  "jhipster:spring-data-couchbase",
  "jhipster:spring-data-elasticsearch",
  "jhipster:spring-data-mongodb",
  "jhipster:spring-data-neo4j",
  "jhipster:spring-data-relational",
  "jhipster:spring-websocket",
  "jhipster:upgrade",
  "jhipster:vue",
  "jhipster:workspaces",
]
`;

exports[`helpers runJHipster defaults should register jhipster generators namespaces 1`] = `
[
  "jhipster:angular",
  "jhipster:app",
  "jhipster:base",
  "jhipster:base-application",
  "jhipster:base-core",
  "jhipster:base-entity-changes",
  "jhipster:base-workspaces",
  "jhipster:bootstrap",
  "jhipster:bootstrap-application",
  "jhipster:bootstrap-application-base",
  "jhipster:bootstrap-application-client",
  "jhipster:bootstrap-application-server",
  "jhipster:bootstrap-workspaces",
  "jhipster:ci-cd",
  "jhipster:client",
  "jhipster:client:common",
  "jhipster:common",
  "jhipster:cucumber",
  "jhipster:cypress",
  "jhipster:docker",
  "jhipster:docker-compose",
  "jhipster:entities",
  "jhipster:entity",
  "jhipster:export-jdl",
  "jhipster:feign-client",
  "jhipster:gatling",
  "jhipster:generate-blueprint",
  "jhipster:git",
  "jhipster:gradle",
  "jhipster:gradle:code-quality",
  "jhipster:gradle:jib",
  "jhipster:gradle:node-gradle",
  "jhipster:heroku",
  "jhipster:info",
  "jhipster:init",
  "jhipster:java",
  "jhipster:java:bootstrap",
  "jhipster:java:build-tool",
  "jhipster:java:code-quality",
  "jhipster:java:domain",
  "jhipster:java:jib",
  "jhipster:java:node",
  "jhipster:java:openapi-generator",
  "jhipster:javascript:bootstrap",
  "jhipster:javascript:eslint",
  "jhipster:javascript:husky",
  "jhipster:javascript:prettier",
  "jhipster:jdl",
  "jhipster:kubernetes",
  "jhipster:kubernetes-helm",
  "jhipster:kubernetes-knative",
  "jhipster:languages",
  "jhipster:liquibase",
  "jhipster:maven",
  "jhipster:maven:code-quality",
  "jhipster:maven:frontend-plugin",
  "jhipster:maven:jib",
  "jhipster:project-name",
  "jhipster:react",
  "jhipster:server",
  "jhipster:spring-boot",
  "jhipster:spring-cache",
  "jhipster:spring-cloud-stream",
  "jhipster:spring-cloud-stream:kafka",
  "jhipster:spring-cloud-stream:pulsar",
  "jhipster:spring-cloud:gateway",
  "jhipster:spring-data-cassandra",
  "jhipster:spring-data-couchbase",
  "jhipster:spring-data-elasticsearch",
  "jhipster:spring-data-mongodb",
  "jhipster:spring-data-neo4j",
  "jhipster:spring-data-relational",
  "jhipster:spring-websocket",
  "jhipster:upgrade",
  "jhipster:vue",
  "jhipster:workspaces",
]
`;

exports[`helpers runJHipster with useDefaultMocks should register jhipster generators namespaces 1`] = `
[
  "jhipster:angular",
  "jhipster:app",
  "jhipster:base",
  "jhipster:base-application",
  "jhipster:base-core",
  "jhipster:base-entity-changes",
  "jhipster:base-workspaces",
  "jhipster:bootstrap",
  "jhipster:bootstrap-application",
  "jhipster:bootstrap-application-base",
  "jhipster:bootstrap-application-client",
  "jhipster:bootstrap-application-server",
  "jhipster:bootstrap-workspaces",
  "jhipster:ci-cd",
  "jhipster:client",
  "jhipster:client:common",
  "jhipster:common",
  "jhipster:cucumber",
  "jhipster:cypress",
  "jhipster:docker",
  "jhipster:docker-compose",
  "jhipster:entities",
  "jhipster:entity",
  "jhipster:export-jdl",
  "jhipster:feign-client",
  "jhipster:gatling",
  "jhipster:generate-blueprint",
  "jhipster:git",
  "jhipster:gradle",
  "jhipster:gradle:code-quality",
  "jhipster:gradle:jib",
  "jhipster:gradle:node-gradle",
  "jhipster:heroku",
  "jhipster:info",
  "jhipster:init",
  "jhipster:java",
  "jhipster:java:bootstrap",
  "jhipster:java:build-tool",
  "jhipster:java:code-quality",
  "jhipster:java:domain",
  "jhipster:java:jib",
  "jhipster:java:node",
  "jhipster:java:openapi-generator",
  "jhipster:javascript:bootstrap",
  "jhipster:javascript:eslint",
  "jhipster:javascript:husky",
  "jhipster:javascript:prettier",
  "jhipster:jdl",
  "jhipster:kubernetes",
  "jhipster:kubernetes-helm",
  "jhipster:kubernetes-knative",
  "jhipster:languages",
  "jhipster:liquibase",
  "jhipster:maven",
  "jhipster:maven:code-quality",
  "jhipster:maven:frontend-plugin",
  "jhipster:maven:jib",
  "jhipster:project-name",
  "jhipster:react",
  "jhipster:server",
  "jhipster:spring-boot",
  "jhipster:spring-cache",
  "jhipster:spring-cloud-stream",
  "jhipster:spring-cloud-stream:kafka",
  "jhipster:spring-cloud-stream:pulsar",
  "jhipster:spring-cloud:gateway",
  "jhipster:spring-data-cassandra",
  "jhipster:spring-data-couchbase",
  "jhipster:spring-data-elasticsearch",
  "jhipster:spring-data-mongodb",
  "jhipster:spring-data-neo4j",
  "jhipster:spring-data-relational",
  "jhipster:spring-websocket",
  "jhipster:upgrade",
  "jhipster:vue",
  "jhipster:workspaces",
]
`;
